### ISSUE TITLE
Keras Model Import API Loss Layer inconsistencies

### ISSUE THREAD

**Issue by akamlani** [deeplearning4j/deeplearning4j#4294]
#### Issue Description
KerasModelImport.importKerasModelAndWeights(modelFileName, enforceTrain) vs 
KerasModelImport.importKerasModelAndWeights(jsonConfig, weightsFile, enforceTrain)

produces different results (one includes a Loss Layer to the tail while the other does not):
Reference the following gist for summary output:
https://gist.github.com/akamlani/3a73df77fc1d3cbbc09101ba560a5efc

In addition, it also is unable to handle the Loss Layer:
Reference the following Keras gist for model architecture:
https://gist.github.com/akamlani/f0371c3f91ad3e809a706b026bbf8ddb
```
KerasModel - Model cannot be trained: output layer outputLayer_a_loss is not an IOutputLayer (no loss function specified)
KerasModel - Model cannot be trained: output outputLayer_a_loss is not an IOutputLayer (no loss function specified)
```
Unable to perform training nor inference, as no Output Layer recognized.

#### Version Information
* 0.9.2-SNAPSHOT (Per 11/12)
* Keras 2.0.7 + TensorFlow 1.3.0
* MacOS
* CPU Based

**Comment by maxpumperla** [comment]
@akamlani I've added your case as unit test, see here:

https://github.com/deeplearning4j/deeplearning4j/blob/4f058e80f1f9cf14d80b2700f3cb66bdb3965f37/deeplearning4j-modelimport/src/test/java/org/deeplearning4j/nn/modelimport/keras/e2e/KerasModelEndToEndTest.java#L293-L312

The warnings you previously encountered have been removed, as we can now more reliably assign Loss layers in DL4J.

For the other part of the problem, if you only provide model json and weights, there's simply no training config to read a loss from. Since that is not obvious to endusers, I've added an error now: 

https://github.com/deeplearning4j/deeplearning4j/blob/4f058e80f1f9cf14d80b2700f3cb66bdb3965f37/deeplearning4j-modelimport/src/main/java/org/deeplearning4j/nn/modelimport/keras/KerasModel.java#L164
